library/src/text/raw.rs

```diff
@@ -255,6 +316,15 @@ fn to_syn(RgbaColor { r, g, b, a }: RgbaColor) -> synt::Color {
     synt::Color { r, g, b, a }
 }
 
+fn num_width(mut num: usize) -> usize {
+    let mut width = 0;
+    while num != 0 {
+        width += 1;
+        num /= 10;
+    }
+    width
+}
```

Comment:
```suggestion
fn num_width(num: usize) -> usize {
    (num.checked_ilog10().unwrap_or_default() + 1) as usize
}
```
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c8474458172314fa97326b703a0dcb92